src/battle_anim_new.c

```diff
@@ -9152,12 +9152,7 @@ void AnimTask_GetWeatherToSet(u8 taskId)
 void AnimTask_SyrupBomb(u8 taskId)
 {
     struct Pokemon *party = GetBattlerParty(gBattleAnimAttacker);
-    u32 isShiny = IsMonShiny(&party[gBattlerPartyIndexes[gBattleAnimAttacker]]);
-
-    if (!(gStatuses4[gBattleAnimTarget] & STATUS4_SYRUP_BOMB))
-        gDisableStructs[gBattleAnimTarget].syrupBombIsShiny = isShiny;
-
-    gBattleAnimArgs[0] = isShiny;
+    gBattleAnimArgs[0] = IsMonShiny(&party[gBattlerPartyIndexes[gBattleAnimAttacker]]);
```

Comment:
Like I commented, `gBattleAnimArgs[0] = gDisableStructs[gBattleAnimTarget].syrupBombIsShiny`

Reply:
I tried that. In this case the attack of a shiny mon was non shiny while the speed drop anim was shiny.

Reply:
Sorry, I'm now noticing you did do that below, this is correct then